torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -213,6 +213,7 @@ void ThreadPredicateMap::updateBitSet(const Expr* expr) {
     // Change for welford Op, we want the users of all outputs of welfordOp
     //  to use a single predicate name.
     if (auto tv_def = tv_inp->definition()) {
+      // TODO: Do we need to do anything for grouped reduction here?
```

Comment:
Is this necessary for `WelfordOp`? The maps of `ThreadPredicateMap` have mappings for all outputs: https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp#L281-L285

Reply:
Commented out this part, and nothing seems to fail.

Reply:
Why are your comments not showing up inline in the files page? Strange.

Reply:
Comes from:
https://github.com/csarofeen/pytorch/pull/561/files#diff-48ec14efa321f9f6f479de4d2c9e377c847067825513a7231d94200d8ea60efaR141-R149

It doesn't seem to be necessarily related to correctness, but just wanting one predicate for all outputs. It's just moving from something like `WelfordResult::var_sum` to be `WelfordResult::avg` so that tv_inp is consistent when you hit:
```
const auto& pred_info = at(tv_inp);
```
If `tv_inp` is the result of a multi output expression, the same `pred_info` comes up for all those siblings.

Reply:
I'm going to keep this logic, but once we cleanup predicate handling based on ID graph we can remove this type of logic.